Windows/MainWindow.GridControls.cs

```diff
@@ -102,20 +106,72 @@ public partial class MainWindow : Window {
             if (!editorIsLoaded) {
                 return;
             }
-            Point mousePos = e.GetPosition(EditorGrid);
-            gridController.GridMouseMove(mousePos);
+            if (editorIsHoldScrolling) {
+                editorIsDeferredHoldScrollingStarted = false; //standard scrolling (Mouse down -> Move)
+
+                var currentPosition = e.GetPosition(scrollEditor);
+                var offset = currentPosition - editorScrollStart.Value;
+                offset.Y /= Editor.HoldScroll.Slowdown;
+
+                if (Math.Abs(offset.Y) > Editor.HoldScroll.DeadZone / Editor.HoldScroll.Slowdown) {
+                    scrollEditor.ScrollToVerticalOffset(scrollEditor.VerticalOffset + offset.Y);
+                    ScrollEditor_ScrollChanged(null, null);
+                }
+            } else {
+                Point mousePos = e.GetPosition(EditorGrid);
+                gridController.GridMouseMove(mousePos);
+
+                // update beat display
+                lblSelectedBeat.Content = $"Time: {Helper.TimeFormat(gridController.snappedBeat * 60 / globalBPM)}, Global Beat: {Math.Round(gridController.snappedBeat, 3)} ({Math.Round(gridController.unsnappedBeat, 3)})";
 
-            // update beat display
-            lblSelectedBeat.Content = $"Time: {Helper.TimeFormat(gridController.snappedBeat * 60 / globalBPM)}, Global Beat: {Math.Round(gridController.snappedBeat, 3)} ({Math.Round(gridController.unsnappedBeat, 3)})";
+                // initiate drag selection
+                if (editorMouseDown) {
+                    Vector delta = mousePos - editorDragSelectStart;
+                    if (delta.Length > Editor.DragInitThreshold) {
+                        gridController.BeginDragSelection(mousePos);
```

Comment:
Inconsistent behavior with the note selection - if user holds the left button before entering this mode, the selection rectangle isn't updated properly when mouse is moved and if left button is released while in this mode, it cancels the scrolling, but the selection mode is still active and can only be cancelled by left click, not selecting any notes.